src/storage/concurrency_manager/lock_table.rs

```diff
@@ -0,0 +1,247 @@
+// Copyright 2020 TiKV Project Authors. Licensed under Apache-2.0.
+
+use super::key_handle::{KeyHandle, KeyHandleGuard};
+
+use parking_lot::Mutex;
+use std::{
+    collections::BTreeMap,
+    ops::Bound,
+    sync::{Arc, Weak},
+};
+use txn_types::{Key, Lock};
+
+#[derive(Clone, Default)]
+pub struct LockTable(pub Arc<Mutex<BTreeMap<Key, Weak<KeyHandle>>>>);
+
+impl LockTable {
+    pub async fn lock_key(&self, key: &Key) -> KeyHandleGuard {
+        loop {
+            if let Some(handle) = self.get(key) {
```

Comment:
It's impossible?

Reply:
Currently we only call `lock_key` when latch is acquired so it's impossible.
In the future we may call it when latch is not acquired (and to replace latch), then it is possible.